storage_v2/spanstore/writer.go

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package spanstore
+
+import (
+	"context"
+
+	"go.opentelemetry.io/collector/pdata/ptrace"
+)
+
+// Writer writes spans to storage.
+type Writer interface {
+	// WriteTrace writes a batch of spans to storage. Idempotent.
+	// Implementations are not required to support atomic transactions,
+	// so if any of the spans fail to be written an error is returned.
+	// Compatible with OTLP Exporter API.
+	WriteTraces(ctx context.Context, td ptrace.Traces) error
```

Comment:
Why no pointer here? If we can use fewer pointers it's better, but then I'd like the reader to use the same type.

Reply:
Added it. But why fewer pointers is better?

Reply:
Passing data by reference requires that data to be heap-allocated, which is usually more expensive than copying the struct (or even a slice of structs) by value.